cmd/query/app/static_handler_test.go

```diff
@@ -117,57 +119,67 @@ func TestNewStaticAssetsHandlerErrors(t *testing.T) {
 func TestHotReloadUIConfigTempFile(t *testing.T) {
-	tmpfile, err := ioutil.TempFile("", "ui-config-hotreload.*.json")
-	assert.NoError(t, err)
+	run := func(description string, extension string) {
+		t.Run(description, func(t *testing.T) {
+			tmpfile, err := ioutil.TempFile("", "ui-config-hotreload.*."+extension)
+			assert.NoError(t, err)
 
-	tmpFileName := tmpfile.Name()
-	defer os.Remove(tmpFileName)
+			tmpFileName := tmpfile.Name()
+			defer os.Remove(tmpFileName)
 
-	content, err := ioutil.ReadFile("fixture/ui-config-hotreload.json")
-	assert.NoError(t, err)
+			content, err := ioutil.ReadFile("fixture/ui-config-hotreload." + extension)
+			assert.NoError(t, err)
 
-	err = ioutil.WriteFile(tmpFileName, content, 0644)
-	assert.NoError(t, err)
+			err = ioutil.WriteFile(tmpFileName, content, 0644)
+			assert.NoError(t, err)
 
-	h, err := NewStaticAssetsHandler("fixture", StaticAssetsHandlerOptions{
-		UIConfigPath: tmpFileName,
-	})
-	assert.NoError(t, err)
+			h, err := NewStaticAssetsHandler("fixture", StaticAssetsHandlerOptions{
+				UIConfigPath: tmpFileName,
+			})
+			assert.NoError(t, err)
+
+			c := string(h.indexHTML.Load().([]byte))
+			assert.Contains(t, c, "About Jaeger")
 
-	c := string(h.indexHTML.Load().([]byte))
-	assert.Contains(t, c, "About Jaeger")
+			newContent := strings.Replace(string(content), "About Jaeger", "About a new Jaeger", 1)
+			err = ioutil.WriteFile(tmpFileName, []byte(newContent), 0644)
+			assert.NoError(t, err)
 
-	newContent := strings.Replace(string(content), "About Jaeger", "About a new Jaeger", 1)
-	err = ioutil.WriteFile(tmpFileName, []byte(newContent), 0644)
-	assert.NoError(t, err)
+			done := make(chan bool)
+			go func() {
+				for {
+					i := string(h.indexHTML.Load().([]byte))
 
-	done := make(chan bool)
-	go func() {
-		for {
-			i := string(h.indexHTML.Load().([]byte))
+					if strings.Contains(i, "About a new Jaeger") {
+						done <- true
+					}
+					time.Sleep(10 * time.Millisecond)
+				}
+			}()
 
-			if strings.Contains(i, "About a new Jaeger") {
-				done <- true
+			select {
+			case <-done:
+				assert.Contains(t, string(h.indexHTML.Load().([]byte)), "About a new Jaeger")
+			case <-time.After(time.Second):
+				assert.Fail(t, "timed out waiting for the hot reload to kick in")
 			}
-			time.Sleep(10 * time.Millisecond)
-		}
-	}()
-
-	select {
-	case <-done:
-		assert.Contains(t, string(h.indexHTML.Load().([]byte)), "About a new Jaeger")
-	case <-time.After(time.Second):
-		assert.Fail(t, "timed out waiting for the hot reload to kick in")
+		})
 	}
+
+	run("json hot reload", "json")
+	run("js hot reload", "js")
 }
 
 func TestLoadUIConfig(t *testing.T) {
 	type testCase struct {
 		configFile    string
-		expected      map[string]interface{}
+		expected      *UIConfigOptions
 		expectedError string
 	}
 
+	jsonRegExpPattern := regexp.MustCompile("JAEGER_CONFIG *= *DEFAULT_CONFIG;")
+	jsRegExpPattern := regexp.MustCompile(`(?im)(^\s+)\/\/.*JAEGER_CONFIG_JS.*\n.*`)
```

Comment:
Could we just use the constants here? Also, do we even need to be comparing which regex is returned? I would rather express the test in terms of what the resulting HTML looks like, not the implementation details of how that HTML is produced.

Reply:
> Could we just use the constants here?

Sure

> Also, do we even need to be comparing which regex is returned? 

This is unit test, I believe, and `LoadUIConfig` is return 
```
type UIConfigOptions struct {
	regexp *regexp.Regexp
	config []byte
}
```
regexp is part of it...


>  I would rather express the test in terms of what the resulting HTML looks like, not the implementation details of how that HTML is produced.

Will update `TestRegisterStaticHandler ` to cover JS file config case